lib/engine/game/g_18_esp/game.rb

```diff
@@ -1185,6 +1245,40 @@ def move_assets(survivor, nonsurvivor)
 
           @log << "Moved assets from #{nonsurvivor.name} to #{survivor.name}"
         end
+
+        def combined_base_trains_candidates(corporation)
+          return unless corporation
+
+          corporation.trains.reject { |t| combined_trains.key?(t) || t.name == '2P' }
+        end
+
+        def combined_obsolete_trains_candidates(corporation)
+          return unless corporation
+
+          rusted_trains = @depot.trains.select do |train|
+            train.rusted && corporation.cash >= 2 * train.price
+          end
+
+          rusted_trains.uniq { |train| train.variants.keys[0] }
+        end
+
+        def update_trains_cache
+          update_cache(:trains)
+        end
+
+        def distance(train)
+          if train.distance.is_a?(Numeric)
+            [train.distance, 0]
+          else
+            cities = train.distance[1]['pay']
+            towns = train.distance[0]['pay']
+            [cities, towns]
+          end
+        end
+
+        def game_corporations
+          self.class::CORPORATIONS + self.class::EXTRA_CORPORATIONS
+        end
```

Comment:
in the next PR, i will add an optional base/beginner version without variable setup.